src/coingecko.ts

```diff
@@ -0,0 +1,220 @@
+import fetch from "node-fetch";
+import fs from "fs";
+import path from "path";
+import { CoinGeckoClient, CoinMarket } from "coingecko-api-v3";
+import { Token, TokenList, Version } from "./types";
+
+const BUILD_DIR = path.join(".", "build");
+const LIST_DIR = path.join(BUILD_DIR, "lists");
+const PAGE_LIMIT = 50;
+const PAGE_COUNT = 5;
+const FILE_PREFIX = "coingecko";
+const OUTPUT_FILE = "CoinGecko.json";
+const SUPPORTED_CHAINS = [
+  { name: "ethereum", chainId: 1 },
+  { name: "xdai", chainId: 100 },
+];
+
+const client = new CoinGeckoClient({
+  timeout: 10_000,
+  autoRetry: true, // will automatically handle 429s and adjust retry time
+});
+
+async function fetchCoingeckoAll() {
+  console.log(`Fetch all CoinGecko's ERC20 Tokens`);
+  const url = "https://tokens.coingecko.com/uniswap/all.json";
```

Comment:
Here we do the query to the API. If we use their library, I would expect they offer to do this same query and already provide the right types